platform/platform.go

```diff
@@ -1,21 +1,34 @@
 package platform
 
 import (
+	"fmt"
+
 	"github.com/buildpacks/lifecycle/api"
-	"github.com/buildpacks/lifecycle/cmd"
-	v05 "github.com/buildpacks/lifecycle/platform/v05"
+	"github.com/buildpacks/lifecycle/platform/common"
+	pre06 "github.com/buildpacks/lifecycle/platform/pre06"
 	v06 "github.com/buildpacks/lifecycle/platform/v06"
+	v07 "github.com/buildpacks/lifecycle/platform/v07"
 )
 
-type Platform interface {
-	API() string
-	CodeFor(errType cmd.LifecycleExitError) int
+var platform03 = pre06.NewPlatform("0.3")
+var platform04 = pre06.NewPlatform("0.4")
+var platform05 = pre06.NewPlatform("0.5")
+var platform06 = v06.NewPlatform(platform05)
+var platform07 = v07.NewPlatform(platform06)
+
+var platformMap = map[string]common.Platform{
+	"0.3": platform03,
+	"0.4": platform04,
+	"0.5": platform05,
+	"0.6": platform06,
+	"0.7": platform07,
 }
 
-func NewPlatform(apiStr string) Platform {
+func NewPlatform(apiStr string) (common.Platform, error) {
 	platformAPI := api.MustParse(apiStr)
-	if platformAPI.Compare(api.MustParse("0.6")) < 0 { // platform API < 0.6
-		return v05.NewPlatform(apiStr)
+	p, ok := platformMap[platformAPI.String()]
```

Comment:
```suggestion
	p, ok := platformMap[apiStr]
```
This would allow you to delete the line above I believe.